src/templating.rs

```diff
@@ -0,0 +1,66 @@
+use serde::de::{Error, Visitor};
```

Comment:
Unlike my broader bias toward end-to-end tests, this one feels ripe for a few tiny unit tests in the module itself :)

Reply:
Haha will do :)

Reply:
Done! :)